iis-core/src/test/java/eu/dnetlib/iis/core/AbstractOozieWorkflowTestCase.java

```diff
@@ -0,0 +1,233 @@
+package eu.dnetlib.iis.core;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.oozie.client.OozieClient;
+import org.apache.oozie.client.OozieClientException;
+import org.apache.oozie.client.WorkflowJob.Status;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+
+/**
+ * Base class for testing oozie workflows on cluster
+ * 
+ * @author madryk
+ *
+ */
+public abstract class AbstractOozieWorkflowTestCase {
+	
+	private final static Logger log = LoggerFactory.getLogger(AbstractOozieWorkflowTestCase.class);
+	
+	
+	private final static String OOZIE_SERVICE_LOC_KEY = "oozieServiceLoc";
+	
+	private final static String OOZIE_RUN_WORKFLOW_LOG_FILE_KEY = "oozie.execution.log.file.location";
+	
+	private final static String OOZIE_WORKFLOW_DEPLOY_MODE_KEY = "deploy.mode";
+	
+	private final static String WORKFLOW_SOURCE_DIR_KEY = "workflow.source.dir";
+	
+	private final static String MAVEN_TEST_WORKFLOW_PHASE = "clean package";
+	
+	private final static String MAVEN_TEST_WORKFLOW_LOCALLY_PROFILE = "attach-test-resources,oozie-package,deploy-local,run-local";
+	
+	private final static String MAVEN_TEST_WORKFLOW_REMOTELY_PROFILE = "attach-test-resources,oozie-package,deploy,run";
+	
+
+	private static IntegrationTestPropertiesReader propertiesReader;
+	
+	private OozieClient oozieClient;
+	
+	
+	private enum DeployMode {
+		LOCAL,
+		SSH
+	}
+	
+	
+	@BeforeClass
+	public static void classSetUp() {
+		propertiesReader = new IntegrationTestPropertiesReader();
+	}
+	
+	@Before
+	public void setUp() {
+		
+		log.debug("Setting up OozieClient at {}", getOozieServiceLoc());
+		oozieClient = new OozieClient(getOozieServiceLoc());
+	}
+	
+	/**
+	 * Tests workflow located under {@literal workflowLocation} parameter.
+	 * Internally uses {@link #testWorkflow(String, OozieWorkflowTestConfiguration)}
+	 * with default {@link OozieWorkflowTestConfiguration}
+	 */
+	protected void testWorkflow(String workflowLocation) {
+		testWorkflow(workflowLocation, new OozieWorkflowTestConfiguration());
+	}
+	
+	/**
+	 * Tests workflow located under {@literal workflowLocation} parameter.
+	 * 
+	 * @param workflowLocation - Folder that contains workflow data.
+	 *  	It must contain file {@literal oozie_app/workflow.xml} with workflow definition.
+	 * @param configuration - some additional test pass criteria
+	 *  	(for example timeout - workflow must finish its execution before specified amount of time, 
+	 *   	otherwise test will automatically fail)
```

Comment:
workflow... before specified amount -> a workflow... within specified time
otherwise test -> otherwise the test